controller/src/main/java/org/jboss/as/controller/operations/common/ProcessEnvironment.java

```diff
@@ -117,6 +130,34 @@ protected abstract boolean isRuntimeSystemPropertyUpdateAllowed(String propertyN
      */
     protected abstract void systemPropertyUpdated(String propertyName, String propertyValue);
 
+    /**
+     * Get the UUID of this process.
+     * @return the UUID of this process.
+     */
+    public abstract UUID getInstanceUuid();
+
+    protected UUID obtainProcessUUID(final Path filePath) throws Exception {
```

Comment:
It probably makes more sense to throw an `IllegalStateException` here as each usage of this method just catches an exception and throws an `IllegalStateException`.